### PR TITLE
Fix null annotation of SyntaxFactory.UsingStatement

### LINKED ISSUE #51924: SyntaxFactory.UsingStatement incorrectly annotates 'declaration' and 'expression' parameters as disallowing null
**Version Used**: 3.9.0 NuGet packages

Some overloads of SyntaxFactory.UsingStatement allow nulls, and the overloads that appear to disallow nulls actually allow them (and must allow them).

For example:
image

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #51924

### CODE REVIEW

**Review by cston — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Syntax/UsingStatementSyntax.cs [hunk 1]
```diff
@@ -20,13 +20,13 @@ namespace Microsoft.CodeAnalysis.CSharp
 {
     public partial class SyntaxFactory
     {
-        public static UsingStatementSyntax UsingStatement(SyntaxToken usingKeyword, SyntaxToken openParenToken, VariableDeclarationSyntax declaration, ExpressionSyntax expression, SyntaxToken closeParenToken, StatementSyntax statement)
+        public static UsingStatementSyntax UsingStatement(SyntaxToken usingKeyword, SyntaxToken openParenToken, VariableDeclarationSyntax? declaration, ExpressionSyntax? expression, SyntaxToken closeParenToken, StatementSyntax statement)
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Syntax/UsingStatementSyntax.cs:23]: Should similar annotations be added to `UsingStatementSyntax.Update` methods above?

**Review by cston — APPROVED** [review]

**Review by RikkiGibson — APPROVED** [review]

**Comment by 333fred** [comment]
@Youssef1313 looks like there's a conflict here.

**Comment by jcouv** [comment]
Can be merged/squashed once CI is green. I just restarted some integration legs.

**Comment by jcouv** [comment]
Merged/squashed. Thanks @Youssef1313 !

**Pull request merged**